corehq/apps/reports/static/reports/js/bootstrap3/base.js

```diff
@@ -24,10 +24,10 @@ hqDefine("reports/js/bootstrap3/base", [
         defaultConfig.datespan_filters = [];
         defaultConfig.datespan_slug = null;
 
-        var $savedReports = $("#savedReports");
+        var $savedReports = $("#savedReports" + initialPageData.get("html_id_suffix"));
```

Comment:
Hey @kaapstorm 

One question to understand this better.

So, since we have multiple reports on the same page, we want the IDs to be unique. I get that.

For the javascript, it would still work on one element only? Or will it be loaded multiple times with ID for each report?

Reply:
That's a good catch - I think the `$savedReports.koApplyBindings(reportConfigsView);` will error, based on this code: https://github.com/dimagi/commcare-hq/blob/891cc2068463782bb606b2699eced654627fca8f/corehq/apps/hqwebapp/static/hqwebapp/js/bootstrap3/hq.helpers.js#L140-L142

Most jQuery functions can handle multiple elements - like [this](https://github.com/dimagi/commcare-hq/blob/891cc2068463782bb606b2699eced654627fca8f/corehq/apps/reports/static/reports/js/bootstrap5/report_config_models.js#L170) usage of `find` will find all inputs in all filter forms - but HQ code isn't necessarily written to handle multiple.

Reply:
Yeah, as I progressed, I realized that using `initialPageData` only makes sense when async reports are rendering HTML for just one report. Otherwise we should pass per-report `options` to the report viewmodels, and use `{{ report.html_id_suffix }}` in templates (which of course will work for async reports too).

Thanks for catching this. I'll remove all instances of `initialPageData.get("html_id_suffix")`.

Reply:
Ah, I hadn't quite put that together (the implications of there only being one initial page data), that's a great way to think about it.